File: prime.js
"use strict";

let n = parseInt(process.argv[2]);
if (n <= 2) {
    console.log("引数は2より大きい数値にしてください:");
    process.exit(0);
}

for (let i = 2; i < n;) {
    if ((n % i) === 0) {
        n = n / i;
        console.log(i);
    } else {
        i++;
    }
    if (n === i) {
        console.log(i);
        break;
    }
}

File: prime.c
#include <stdio.h>
#include <stdlib.h>
int main(int argc, char *argv[]){
	int n;
	if(argc < 2){
		printf("自然数を入力してください\n");
		scanf("%d", &n);
	}else{
		n = atoi(argv[1]);
		printf("%d\n", n);
	}
	if(n <= 2){
		printf("2より大きい自然数を入力してください\n");
		exit(0);
	}
	for(int i = 2; i < n;){
		if((n % i) == 0){
			n = n / i;
			printf("%d\n", i);
		}else{
			i++;
		}
		if(n == i){
			printf("%d\n", i);
			break;
		}
	}
	return 0;
}

